specs/vep-1722-properties-based-vdk-configuration-provider/README.md

```diff
@@ -0,0 +1,149 @@
+
+# VEP-1722: Properties based VDK Configuration Provider
+
+* **Author(s):** Antoni Ivanov (aivanov@vmware.com),
+* **Status:** draft
+
+
+To get started with this template:
+
+- [ ] **Create an issue in Github (if one does not exists already)**
+- [ ] **Make a copy of this template directory.**
+  Copy this template into the specs directory and name it
+  `NNNN-short-descriptive-title`, where `NNNN` is the issue number (with no
+  leading-zero padding) created above.
+- [ ] **Fill out this file as best you can.**
+  There are instructions as HTML comments.
+  At minimum, you should fill in the "Summary" and "Motivation" sections.
+- [ ] **Create a PR for this VEP.**
+- [ ] **Merge early and iterate.**
+  Avoid getting hung up on specific details and instead aim to get the goals of
+  the VEP clarified and merged quickly. The best way to do this is to just
+  start with the high-level sections and fill out details incrementally in
+  subsequent PRs.
+
+<!-- Provide table of content as it's helpful. -->
+
+- [Summary](#summary)
+- [Glossary](#glossary)
+- [Motivation](#motivation)
+- [Requirements and goals](#requirements-and-goals)
+- [High-level design](#high-level-design)
+- [API Design](#api-design)
+- [Detailed design](#detailed-design)
+- [Implementation stories](#implementation-stories)
+- [Alternatives](#alternatives)
+
+## Summary
+
+This VEP proposes a new provider for VDK configuration that allows users to set job-specific configuration for data jobs.
+Currently, users can only set configuration using environment variables or the config.ini file, both of which have limitations.
+The new provider will take configuration from properties with the "vdk.config." prefix and have lesser priority than environment variables and the config.ini file.
+
+## Glossary
+<!--
+Optional section which defines terms and abbreviations used in the rest of the document.
+-->
+
+## Motivation
+VDK allows users to develop and execute data jobs.
+Each data job can require its own specific configuration that needs to be set before execution.
+Currently, the only way to set configuration is using environment variables or the config.ini file.
+However, these approaches have limitations. Users cannot set environment variables for deployed jobs in the cloud, and the config.ini file is not secure for storing sensitive data.
+
+To address these limitations, we propose a new provider for VDK configuration that allows users to set job-specific configuration using properties with the "vdk.config." prefix.
+This approach provides more flexibility and security in setting configuration, especially for sensitive data.
+
+The current limitations in setting job-specific configuration within VDK can also impact the system's usefulness for users.
+One issue is the lack of flexibility, as users are unable to adopt a new database plugin unless it has been pre-configured by a system administrator.
```

Comment:
That is a very good point. The argumentation is well made.
nit: my suggestion is to include link to Github issue if available

Reply:
add the issue link on the top. Though the VEP number is the issue number.